moveit_planners/pilz_industrial_motion_planner/src/trajectory_functions.cpp

```diff
@@ -72,10 +72,7 @@ bool pilz_industrial_motion_planner::computePoseIK(const planning_scene::Plannin
     return false;
   }
 
-  moveit::core::RobotState rstate(robot_model);
-  // By setting the robot state to default values, we basically allow
-  // the user of this function to supply an incomplete or even empty seed.
-  rstate.setToDefaultValues();
```

Comment:
It's also problematic that we can't check for a complete state here since there is no access to the PSM. Maybe we should enforce this more on the runtime interface side: MoveItCpp/MoveGroup could run "CurrentStateMonitor::waitForCompleteState()" by default.

Reply:
> As stated here, we would drop support for incomplete states by removing this line. This is not necessarily a blocker, though. But I agree that it would be unsafe to assume that default values are given as long as the other groups are not initialized. If the user wants to ignore other groups, this should be done by providing a fixed ACM. We should check that this info is not lost or that it won't conflict with the tutorials.

I think that we are still supporting incomplete and even empty seed with the only difference that joint values not specified are set from the current robot state instead of the default one.


> It's also problematic that we can't check for a complete state here since there is no access to the PSM. Maybe we should enforce this more on the runtime interface side: MoveItCpp/MoveGroup could run "CurrentStateMonitor::waitForCompleteState()" by default.

Totally agree with you on this point.

Reply:
Can we open follow-up issues for this?

Reply:
yes, please do